node-registrar/pkg/db/models.go

```diff
@@ -0,0 +1,59 @@
+package db
+
+type Farm struct {
+	FarmID      uint64 `gorm:"primaryKey;autoIncrement" json:"farm_id"` // Primary key
+	FarmName    string `gorm:"size:100;not null;unique" json:"farm_name"`
+	TwinID      uint64 `json:"twin_id" gorm:"not null"`
+	Dedicated   bool   `json:"dedicated"`
+	FarmFreeIps uint64 `json:"farm_free_ips"`
+
+	Nodes []Node `gorm:"foreignKey:farm_id;constraint:OnDelete:CASCADE" json:"nodes"`
+}
+
+type Node struct {
+	NodeID      uint64      `gorm:"primaryKey;autoIncrement" json:"node_id"`
+	FarmID      uint64      `gorm:"not null" json:"farm_id"`
+	TwinID      uint64      `json:"twin_id" gorm:"not null"`
+	Features    []string    `gorm:"type:jsonb;serializer:json"`
+	PriceUsd    float64     `json:"price_usd"`
+	ExtraFee    uint64      `json:"extra_fee"`
+	Status      string      `gorm:"size:50" json:"status"`
+	Healthy     bool        `json:"healthy"`
+	Dedicated   bool        `json:"dedicated"`
+	Rented      bool        `json:"rented"`
+	Rentable    bool        `json:"rentable"`
+	Uptime      Uptime      `json:"uptime"`
```

Comment:
You are currently storing only the latest uptime report from nodes. How this would be useable? Shouldn't instead support storing multiple uptime records for the same node over time. In this case, it would be more appropriate to use a separate table specifically for uptime reports.